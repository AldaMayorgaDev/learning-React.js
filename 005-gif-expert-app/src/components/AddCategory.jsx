import { useState } from 'react';
import {PropTypes} from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    //console.log("event", event);
    //console.log("inputValue", inputValue);

    if(inputValue.trim().length <=1) return;
    


    //setCategoriesProp(categories=>[inputValue, ...categories]);
    setInputValue('');
    onNewCategory(inputValue.trim())

   
  };
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        placeholder="Buscar Gifs 🔍"
        type="text"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};


AddCategory.propTypes ={
  onNewCategory: PropTypes.func.isRequired,
};