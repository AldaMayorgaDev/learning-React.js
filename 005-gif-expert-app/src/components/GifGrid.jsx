
import {PropTypes} from 'prop-types';
import {GifItem, Sppiner} from './';
import { useFetchGifs } from '../hooks/useFetchGifs';





export const GifGrid = ({categoria}) => {

  const {imagenes, isLoading} = useFetchGifs (categoria);
  
  return (
    <>
        <h3>{categoria}</h3>

      {/*   {
          isLoading ? (<Sppiner/>): null
        } */}

       <Sppiner isLoading={isLoading}/>
        <div className="card-grid">
          {imagenes.map((imagen)=>{
            const {id} = imagen;

            return (
              <GifItem 
                key={id}
                {...imagen} //se envia todas las propiedades de imagen como props (se esparse)
              />
            );
          })}
        </div>
 </>
    
  )
}

GifGrid.propTypes={
    categoria: PropTypes.string.isRequired,
}