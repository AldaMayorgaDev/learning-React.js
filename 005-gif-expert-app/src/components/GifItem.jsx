import {PropTypes} from 'prop-types';
export const GifItem = (imagen) => {
    const {id, titulo, url} = imagen
  return (
    <div className="card" id={id}>
        <img src={url} alt={titulo} title={titulo} />
        <p>{titulo}</p>
    </div>
  )
}

GifItem.propTypes = {
  titulo: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.string
};

