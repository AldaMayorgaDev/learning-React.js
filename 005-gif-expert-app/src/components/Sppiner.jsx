import { PropTypes } from 'prop-types';
export const Sppiner = ({isLoading}) => {
  if(isLoading){
      return (
    <>
      <div className="sk-chase">
        <div className="sk-chase-dot"></div>
        <div className="sk-chase-dot"></div>
        <div className="sk-chase-dot"></div>
        <div className="sk-chase-dot"></div>
        <div className="sk-chase-dot"></div>
        <div className="sk-chase-dot"></div>
      </div>
    </>
  );
  }
}

Sppiner.propTypes={
  isLoading: PropTypes.bool.isRequired
}