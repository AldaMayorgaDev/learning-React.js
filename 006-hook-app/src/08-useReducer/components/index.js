export * from './ToDoItem.jsx';
export * from './ToDoList.jsx';
export * from './TodoAdd.jsx';
export * from './utilities/ContadorToDo.jsx';