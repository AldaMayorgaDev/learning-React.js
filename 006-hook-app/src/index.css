body{
    padding: 20px;
}

button{
    margin-right: 10px;
}

/* a{
    color: #131313;
    text-decoration: none;
}

a:active{
    color: #69696A;
    text-decoration: none;
} */

/* .navbar-brand, .nav-link{
    padding-left: 10px
} */