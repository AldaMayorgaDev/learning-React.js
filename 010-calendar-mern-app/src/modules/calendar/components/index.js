export * from './NavBar/NavBar';
export * from './CalendarEventBox/CalendarEventBox';
export * from './CalendarModal/CalendarModal';
export * from './CalendarModal/FormModal/FormModal';
export * from './FabAddNewEvent/FabAddNewEvent';
export * from './FabDeleteEvent/FabDeleteEvent';