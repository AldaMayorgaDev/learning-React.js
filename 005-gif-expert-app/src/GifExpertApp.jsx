import { useState } from 'react';
import { AddCategory, GifGrid } from './components';


export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Barcelona']);

  const onAddCategory=(newCategory)=>{
    //setCategories(['r',...categories]);

    //Validar que el valor tecleado no este en el areglo categories para no tenerl el problemas de keys

    if(categories.includes(newCategory)){
      //console.log(`${ newCategory } Ya existe...`)
      return; //Se sale
    }
       //console.log('newCategory', newCategory)
       setCategories([ newCategory ,...categories]);
    
   
  }

  return (
    <>
      {/* Titulo */}
      <h1>GifExpertApp</h1>

      {/* Input */}
      <AddCategory
        /* setCategoriesProp={setCategories} */
        onNewCategory={(value) => onAddCategory(value)}
      />

      {/* Listado de gifs */}

        {categories.map((category) => {
          return (
            <GifGrid 
              key={category} 
              categoria={category}  
            />     
          );
        })}
      
    </>
  );
}
