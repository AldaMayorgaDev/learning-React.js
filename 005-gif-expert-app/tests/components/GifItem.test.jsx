import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';
// eslint-disable-next-line no-undef
describe('Pruebas en <GifItem />', ()=>{
    const titulo = 'Legend of zelda';
    const url = 'https://google.com.mx/';

    test('Debe hacer match con el snapshot', () => { 
        const {container}= render(<GifItem titulo ={titulo} url = {url}/>);

        expect(container).toMatchSnapshot();
    });


    test('Debe de mostrar la imagen con el url y el alt inicado', () => { 
        render(<GifItem titulo ={titulo} url={url}/>);
        //render(<GifItem titulo ={titulo} url = {url}/>);
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(titulo);
        const {src, alt} =  screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(titulo);

    });

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifItem titulo ={titulo} url={url}/>);

        expect(screen.getAllByText (titulo).toBeTruthy);
    })
})