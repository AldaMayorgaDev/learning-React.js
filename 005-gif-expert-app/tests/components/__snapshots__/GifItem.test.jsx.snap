// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Legend of zelda"
      src="https://google.com.mx/"
      title="Legend of zelda"
    />
    <p>
      Legend of zelda
    </p>
  </div>
</div>
`;
